contracts/Vault2.sol

```diff
@@ -1,6 +1,24 @@
 //SPDX-License-Identifier: Unlicense
 pragma solidity ^0.8.4;
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import "hardhat/console.sol";
 
-contract Vault2 {
-    
+contract Vault2 is ERC20 {
+
+    constructor () ERC20("Vault", "VAULT") {
+    }
+
+    function mint(uint _amount) public payable {
+        require(_amount > 0, "Can't mint zero");
+        require(msg.value == _amount*(1 ether), "Mint amount differs from sent amount");
```

Comment:
`msg.value` is in wei so better to consume `_amount` also in wei rather than ether